### PR TITLE
feat: new mock plugin to support mocking arbitrary accelerator

### PR DESCRIPTION
**Pull request by kyujin-cho**
This PR adds support to mock any kind of accelerators by updating `cuda-mock` implementations to let user configure slot name of mocked accelerator plugin. Consult `configs/accelerator/mock-accelerator.toml` for more information.

----
:books: Documentation preview :books:: https://sorna--1158.org.readthedocs.build/en/1158/

----
:books: Documentation preview :books:: https://sorna-ko--1158.org.readthedocs.build/ko/1158/

### CODE REVIEW

**Comment by kyujin-cho** [comment]
Notice: once this PR is merged, every users currently utilizing `cuda-mock` plugin need to rewrite their `cuda-mock.toml` config file to match new format.

**Review by achimnol — APPROVED** [review]
LGTM!

**Comment by achimnol** [comment]
This should have a changelog, though.

**Comment by achimnol** [comment]
I'll add some documentation for this.

**Comment by achimnol** [comment]
TODO: rewrite plugin to recognize the new subtype-aware formatting config / update docs accordingly

**Review by achimnol — CHANGES_REQUESTED** [review]
Need to rewrite the plugin to load per-subtype (e.g., `.devices` or `.shares`) format information instead of the global ones.

File: src/ai/backend/accelerator/mock/plugin.py [hunk 1]
```diff
@@ -173,12 +186,16 @@ async def init(self, context: Any = None) -> None:
...
+        for format_name, format in self.mock_config["formats"].items():
+            self.device_formats[format_name] = _format_config_iv.check(format)
+
         if self._mode == AllocationModes.DISCRETE:
             self.slot_types.append((f"{self.key}.device", "count"))  # type: ignore  # (only updated here)
         elif self._mode == AllocationModes.FRACTIONAL:
             self.slot_types.append((f"{self.key}.shares", "count"))  # type: ignore  # (only updated here)
         else:
-            log.error("Not implemented allocation mode: {}", self._mode)
+            log.error("Unimplemented allocation mode: {}", self._mode)
```
Comment (achimnol) [inline src/ai/backend/accelerator/mock/plugin.py]: How about just saying "Invalid"? 😉

**Review by achimnol — COMMENTED** [review]
Added note about just a minor expression. Otherwise, LGTM.

**Pull request merged**